src/modules/FunModule.h

```diff
@@ -2,12 +2,193 @@
+
+				for (auto& option : settings->options) 
+				{
+					double val = static_cast<double>(option.count)/ static_cast<double>(total) * EMBED_BAR_MAX_WIDTH;
+					double percentage = static_cast<double>(option.count) / static_cast<double>(total) * 100.0;
+					int num = qFloor(val);
+
+					QString str{ utility::consts::ZERO_WIDTH_SPACE };
+
+					for (int i = 0; i < num; i++) 
+					{
+						str += utility::consts::emojis::GREEN_BLOCK;
+					}
+
+					for (int i = 0; i < EMBED_BAR_MAX_WIDTH - num;i++) 
+					{
+						str += utility::consts::emojis::BLACK_BLOCK;
+					}
+
+					Discord::EmbedField field;
+					QRegExp reg{ ".+:\\d+" };
+					
+					QString desc = "";
+					QString emoji = option.emote;
+					
+					if (option.desc != "") 
+					{
+						desc = " (" + option.desc + ") ";
+					}
+
+					if (reg.exactMatch(option.emote)) 
+					{
+						if (option.isAnimated) 
+						{
+							emoji.prepend("<a:");
+						}
+						else emoji.prepend("<:");
+						emoji.append(">");
+					}
+					field.setName(emoji + desc + ": " + QString::number(percentage) + "%");
+					field.setValue(str);
+
+					fields.push_back(field);
+				}
+
+				Discord::Embed embed;
+				embed.setColor(qrand() % 16777216);
+				embed.setTitle("Results for Poll#" + QString::number(poll_num) + " " + poll_name);
+				embed.setFields(fields);
+
+
+				UmikoBot::Instance().createMessage(notif_chan, embed).then([poll_num, serverPolls](const Discord::Message& message) 
+				{
+
+					auto settings = serverPolls->at(poll_num);
+					auto poll_name = settings->pollName;
+					auto notif_chan = settings->notifChannel;
+					auto poll_msg = settings->pollMsg;
+
+					//! Get the id of the resulting message
+
+					auto resultMsg = message.id();
+
+					//! This part is used to get the guild id
+
+					UmikoBot::Instance().getChannel(notif_chan).then([poll_num, resultMsg, serverPolls](const Discord::Channel& chan) 
+					{
+
+						auto settings = serverPolls->at(poll_num);
+						auto poll_name = settings->pollName;
+						auto notif_chan = settings->notifChannel;
+						auto poll_msg = settings->pollMsg;
+
+						Discord::MessagePatch patch;
+
+						Discord::Embed embed;
+						embed.setColor(qrand() % 16777216);
+						embed.setTitle("Poll#" + QString::number(poll_num) + " " + poll_name);
+						embed.setDescription("This poll has ended. See the results [here.](https://discordapp.com/channels/" 
+							+ QString::number(chan.guildId()) + "/" 
+							+ QString::number(notif_chan) + "/" 
+							+ QString::number(resultMsg) + ")");
+
+						patch.setEmbed(embed);
+						UmikoBot::Instance().editMessage(notif_chan, poll_msg, patch);
+						UmikoBot::Instance().deleteAllReactions(chan.id(), poll_msg);
+
+						//! Remove the poll from our pending list
+						for (int i = 0; i < serverPolls->size(); i++) 
+						{
+							if (serverPolls->at(i)->pollNum == poll_num)
+							{
+								serverPolls->removeAt(i);
+							}
+						}
+					});
+				});
+			});
+		}
+	};
```

Comment:
I'd prefer it if this was located in the .cpp-file. And since it seems to have a lot of logic behind it, probably its own source files as well.

Reply:
Sure